src/components/signup/EmailConfirmation.js

```diff
@@ -1,42 +1,88 @@
+import GDStore from '../../lib/undux/GDStore'
+import LoadingIndicator, { setLoadingWithStore } from '../common/LoadingIndicator'
+import { Description, LinkButton, NextButton, Title, Wrapper } from './components'
+
 type Props = {
-  screenProps: any
+  screenProps: any,
+  navigation: any
 }
-type State = {}
-export default class EmailConfirmation extends React.Component<Props, State> {
-  handleSubmit = () => {
-    this.props.screenProps.doneCallback({ isEmailConfirmed: true })
+
+const log = logger.child({ from: 'EmailConfirmation' })
+
+const EmailConfirmation = ({ navigation, screenProps }: Props) => {
+  const [globalProfile, setGlobalProfile] = useState({})
+  const API = useWrappedApi()
+  const userStorage = useWrappedUserStorage()
+  const setLoading = setLoadingWithStore(GDStore.useStore())
+
+  useEffect(() => {
+    const { params } = navigation.state
+
+    const validateEmail = async () => {
+      setLoading(true)
+
+      const profile = await userStorage.getProfile()
+      setGlobalProfile(profile)
+
+      await API.verifyEmail({ code: params.validation })
+
+      screenProps.doneCallback({ ...profile, isEmailConfirmed: true })
+    }
+
+    if (params && params.validation) {
+      validateEmail()
+        .then(() => setLoading(false))
+        .catch(() => setLoading(false))
+    }
+  }, [])
+
+  const handleResend = async () => {
+    setLoading(true)
+
+    const profile = globalProfile.email ? globalProfile : screenProps.data
+    await API.sendVerificationEmail(profile)
+
+    setLoading(false)
+  }
+
+  const handleSubmit = () => {
+    log.info('opening email client...')
   }
-  render() {
-    return (
+
+  return (
+    <>
       <Wrapper
-        handleSubmit={this.handleSubmit}
-        footerComponent={props => {
-          return (
-            <React.Fragment>
-              <Description style={styles.description}>{'Please click the link to approve the email'}</Description>
-              <NextButton valid={true} handleSubmit={props.handleSubmit}>
-                Open your email app
-              </NextButton>
-              <LinkButton onPress={() => console.log('Link button')}>Send mail again</LinkButton>
-            </React.Fragment>
-          )
-        }}
+        handleSubmit={handleSubmit}
+        footerComponent={props => (
+          <React.Fragment>
+            <Description style={styles.description}>{'Please click the link to approve the email'}</Description>
+            <NextButton valid={true} handleSubmit={props.handleSubmit}>
+              Open your email app
+            </NextButton>
+            <LinkButton onPress={handleResend}>Send mail again</LinkButton>
+          </React.Fragment>
+        )}
       >
         <Description>{"We've sent an email to:"}</Description>
-        <Title>{this.props.screenProps.data.email}</Title>
+        <Title>{globalProfile.email || screenProps.data.email}</Title>
       </Wrapper>
-    )
-  }
+      <LoadingIndicator />
```

Comment:
I think `LoadingIndicator` component should be in AppSwitch or a global view component, because we probably will use it from most of the pages, and we can handle visibility with undux state, but we don't need to specifically add it in each page...

Reply:
done!